src/store/hooks/useStateStore.ts

```diff
@@ -1,7 +1,10 @@
-import { useEffect, useState } from 'react';
+import { useCallback, useMemo, useSyncExternalStore } from 'react';
```

Comment:
To support older React versions, it's better to use the `use-sync-external-store/shim` package. It also has a `useSyncExternalStoreWithSelector` helper hook which will do selector and selection memoization for you.

Reply:
Thank you, I'll use the shim but the one with selector still requires you to provide a comparator for the selected values so I'll keep it the way it is as it seems that re-writing it to fit the other API would yield similar results - or as would James Franco say:

![](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExb3d0cDR6cjJ0NTZ3ZTQ4NmI0YjJ3ZmFvZTNkaW9tbzZrbHV1cTFpNiZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/C6JQPEUsZUyVq/giphy.gif)

Reply:
Do I understand it well, that not using `use-sync-external-store/shim` could lead to new GH issues of those, who are using React 16 @myandrienko ?

Reply:
For the ones who have React version smaller than React 18.

Reply:
That's correct, you should use `use-sync-external-store/shim` if you want to support `react@<18`. It uses the "native" implementation on `react@>=18`, so there's no real downside.